compiler-cli/src/run.rs

```diff
@@ -21,24 +23,71 @@ pub fn command(
     arguments: Vec<String>,
     target: Option<Target>,
     runtime: Option<Runtime>,
+    module: Option<String>,
     which: Which,
 ) -> Result<(), Error> {
+    // Validate the module to make sure it is a gleam module path
+    match &module {
+        Some(module_name) => {
+            // TODO: Check if this can be replaced with a function that
+            // someone already wrote
+            if !module_name
+                .chars()
+                .all(|x| "abcdefghijklmnopqrstuvwxyz/_".contains(x))
+            {
+                Err(Error::InvalidModuleName {
+                    module: module_name.to_owned(),
+                })
+            } else {
+                Ok(())
+            }
+        }
+        None => Ok(()),
+    }?;
+
     let config = crate::config::root_config()?;
 
     // Determine which module to run
-    let module = match which {
+    let module = module.unwrap_or(match which {
         Which::Src => config.name.to_string(),
         Which::Test => format!("{}_test", &config.name),
-    };
+    });
 
     // Build project so we have bytecode to run
-    let _ = crate::build::main(Options {
+    let (compiler, _) = crate::build::main(Options {
         warnings_as_errors: false,
         codegen: Codegen::All,
         mode: Mode::Dev,
         target,
     })?;
 
+    // A module can not be run if it does not exist or does not have a public main function.
+    match compiler
+        .get_importable_modules()
+        .get(&SmolStr::from(module.to_owned()))
+    {
+        Some(module_data) => match module_data.values.get("main") {
+            Some(ValueConstructor {
+                public: _,
+                variant:
+                    ValueConstructorVariant::ModuleFn {
+                        name: _,
+                        field_map: _,
+                        module: _,
+                        arity: _,
```

Comment:
We'll need to also check that it doesn't take any arguments as otherwise it'll crash when we fail to supply them

Reply:
I added a check for arity with a new error message.